pkg/cmd/pulumi/convert.go

```diff
@@ -120,34 +119,37 @@ func pclGenerateProject(directory string, project workspace.Project, p *pcl.Prog
 // pclEject
 func pclEject(directory string, loader schema.ReferenceLoader) (*workspace.Project, *pcl.Program, error) {
 	parser := hclsyntax.NewParser()
-	// Load all .pp files in the directory
-	err := filepath.WalkDir(directory, func(path string, d fs.DirEntry, err error) error {
-		if d.IsDir() {
-			return nil
+	// Load all .pp files in the components' subdirectory
+	files, err := os.ReadDir(directory)
+	for _, file := range files {
+		if file.IsDir() {
+			continue
 		}
+		fileName := file.Name()
+		path := filepath.Join(directory, fileName)
 
 		if filepath.Ext(path) == ".pp" {
 			file, err := os.Open(path)
 			if err != nil {
-				return err
+				return nil, nil, err
 			}
 
 			err = parser.ParseFile(file, filepath.Base(path))
 			if err != nil {
-				return err
+				return nil, nil, err
 			}
 			diags := parser.Diagnostics
 			if diags.HasErrors() {
-				return diags
+				return nil, nil, diags
 			}
 		}
-		return nil
-	})
+	}
+
 	if err != nil {
 		return nil, nil, err
 	}
 
-	program, pdiags, err := pcl.BindProgram(parser.Files, pcl.Loader(loader))
+	program, pdiags, err := pcl.BindProgram(parser.Files, pcl.Loader(loader), pcl.DirPath(directory))
```

Comment:
Could we switch to passing a https://github.com/spf13/afero file system object rather than a path. Should make some testing easier and is easy to turn a literal path into an afero file system for the final use cases of this.